### ISSUE TITLE
Update code base to a current Terraform version

### ISSUE THREAD

**Issue by agold-rh** [terraform-google-modules/terraform-example-foundation#1265]
### TL;DR

The documents (README) for this code base indicates that Terraform version 1.3.10 should be used to for execution. Current Terraform is 1.8.4 as of this writing, and 1.3.10 is a version no longer supported by Hashicorp. I'm listing this as a bug, because creating new code for an unsupported tf version that's reached EOL is problematic. 

### Expected behavior

I expect most currently supported Terraform versions to execute correctly.
![Screenshot 2024-06-04 at 15 18 57](https://github.com/terraform-google-modules/terraform-example-foundation/assets/41129777/f7d297e8-ff36-4a9a-8504-a1e7a4ba69ec)

### Observed behavior

![Screenshot 2024-06-04 at 15 17 45](https://github.com/terraform-google-modules/terraform-example-foundation/assets/41129777/a7689a7c-949e-4825-a0a0-f6e5ec734e0f)

### Terraform Configuration

```hcl
None
```

### Terraform Version

```sh
None
```

### Additional information

A requirement to use an obsolete version of TF for correct execution is a red flag for any cautious, security conscious software engineer. Since this repo was created to conveniently provide a security conscious landing zone, a TF update seems desirable.

**Comment by fmichaelobrien** [comment]
FYI for cloud build users

An upgrade past 1.3.10 in the DockerFile used in cloud build will benefit from the upgrade - for now we downgrade to 1.3 to avoid a statefile corruption between 0-bootstrap and 1-org
https://github.com/terraform-google-modules/terraform-example-foundation/issues/1151
workaround
https://github.com/GoogleCloudPlatform/pbmm-on-gcp-onboarding/issues/374

**Comment by bdashrad** [comment]
I've been running my own clone of this on terraform 1.8 on github after modifying the deploy scripts.

**Comment by eeaton** [comment]
We've partially updated the version to 1.5.7 (last version before hashicorp changed the license). However, I acknowledge that this doesn't entirely address this issue, because versions 1.6 and earlier are out of support.

I'm seeking legal confirmation internally whether we can use versions after 1.6 with the BSL license changes from Hashicorp. In the meantime we've added text emphasizing that users of the blueprint can upgrade terraform to the latest version so long as they ensure that the TF version used in manual steps and in build pipeline are the same.
